### ISSUE TITLE
Wrong response authentication request when use JWT 

### ISSUE THREAD

**Issue by twelve34** [jhipster/generator-jhipster#3353]
##### **Overview of the issue**

When using JWT authentication and using the wrong credentials a "401 Unauthorized" is replied, sounds oke, but the body is the text _Bad credentials_ and the response header _Content-Type:application/json;charset=UTF-8_

The angular frontend is parsing it as json which results into:

~~~
angular.js:13294 SyntaxError: Unexpected token B
    at Object.parse (native)
    at fromJson 
~~~

##### **JHipster Version(s)** 
3.0.0

##### **Reproduce the error** 

Alway reproducable when sending wrong credentials in signin screen

##### **Suggest a Fix** 

Send json object with exception 
The response is generated into the UserJWTController.java 

~~~java
        try {
            Authentication authentication = this.authenticationManager.authenticate(authenticationToken);
            SecurityContextHolder.getContext().setAuthentication(authentication);
            boolean rememberMe = (loginDTO.isRememberMe() == null) ? false : loginDTO.isRememberMe();
            String jwt = tokenProvider.createToken(authentication, rememberMe);
            response.addHeader(JWTConfigurer.AUTHORIZATION_HEADER, "Bearer " + jwt);
            return ResponseEntity.ok(new JWTToken(jwt));
        } catch (AuthenticationException exception) {
            return new ResponseEntity<>(exception.getLocalizedMessage(), HttpStatus.UNAUTHORIZED);
        }
~~~

**Comment by deepu105** [comment]
@twelve34 would you consider submitting a PR to fix this?

**Comment by twelve34** [comment]
@deepu105  PR done #3371

**Comment by wmarques** [comment]
Closed by #3371 
@twelve34 please add "Fix #issuenumber" next time in our PR, it will automatically close the issue related to the PR :)
